src/tests/src/solver/expressions/test_expressions.cpp

```diff
@@ -0,0 +1,579 @@
+                     bdata::make(LinearStatus_ALL) ^ bdata::make(LinearStatus_ALL),
+                     x,
+                     y)
+{
+    BOOST_CHECK_EQUAL(x + y, y + x);
+}
+
+BOOST_DATA_TEST_CASE(linear_subtract_same_as_plus,
+                     bdata::make(LinearStatus_ALL) ^ bdata::make(LinearStatus_ALL),
+                     x,
+                     y)
+{
+    BOOST_CHECK_EQUAL(x - y, x + y);
+}
+
+BOOST_DATA_TEST_CASE(linear_multiply_commutative,
+                     bdata::make(LinearStatus_ALL) ^ bdata::make(LinearStatus_ALL),
+                     x,
+                     y)
+{
+    BOOST_CHECK_EQUAL(x * y, y * x);
+}
+
+BOOST_FIXTURE_TEST_CASE(simple_linear, Registry<Node>)
+{
+    LiteralNode literalNode1(10.);
+    VariableNode var1("x");
+    // 10.*x
+    Node* u = create<MultiplicationNode>(&literalNode1, &var1);
+
+    LiteralNode literalNode2(20.);
+    ComponentVariableNode var2("id", "y");
+    // 20.*id.y
+    Node* v = create<MultiplicationNode>(&literalNode2, &var2);
+    // 10.*x+20.*id.y
+    Node* expr = create<AddNode>(u, v);
+
+    PrintVisitor printVisitor;
+    BOOST_CHECK_EQUAL(printVisitor.dispatch(*expr), "((10.000000*x)+(20.000000*id.y))");
+    LinearityVisitor linearVisitor;
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*expr), LinearStatus::LINEAR);
+}
+
+BOOST_FIXTURE_TEST_CASE(simple_not_linear, Registry<Node>)
+{
+    VariableNode var1("x");
+    ComponentVariableNode var2("id", "y");
+    // x*id.y
+    Node* expr = create<MultiplicationNode>(&var1, &var2);
+
+    PrintVisitor printVisitor;
+    BOOST_CHECK_EQUAL(printVisitor.dispatch(*expr), "(x*id.y)");
+    LinearityVisitor linearVisitor;
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*expr), LinearStatus::NON_LINEAR);
+}
+
+BOOST_FIXTURE_TEST_CASE(simple_linear_division, Registry<Node>)
+{
+    VariableNode var1("x");
+    // constant
+    ParameterNode param("y");
+    // x/y
+    Node* expr = create<DivisionNode>(&var1, &param);
+
+    PrintVisitor printVisitor;
+    BOOST_CHECK_EQUAL(printVisitor.dispatch(*expr), "(x/y)");
+    LinearityVisitor linearVisitor;
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*expr), LinearStatus::LINEAR);
+}
+
+BOOST_FIXTURE_TEST_CASE(simple_non_linear_division, Registry<Node>)
+{
+    VariableNode var1("x");
+    // variable
+    VariableNode var2("y");
+    // x/y
+    Node* expr = create<DivisionNode>(&var1, &var2);
+
+    PrintVisitor printVisitor;
+    BOOST_CHECK_EQUAL(printVisitor.dispatch(*expr), "(x/y)");
+    LinearityVisitor linearVisitor;
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*expr), LinearStatus::NON_LINEAR);
+}
+
+BOOST_FIXTURE_TEST_CASE(comparison_nodes_are_not_linear, Registry<Node>)
+{
+    PrintVisitor printVisitor;
+    LinearityVisitor linearVisitor;
+
+    VariableNode var1("x");
+    // variable
+    VariableNode var2("y");
+    // x==y
+    Node* eq = create<EqualNode>(&var1, &var2);
+    BOOST_CHECK_EQUAL(printVisitor.dispatch(*eq), "x==y");
```

Comment:
`BOOST_CHECK_EQUAL(printVisitor.dispatch(*eq), "x==y");` : 
Is "==" supposed to be part of a expression ?
Otherwise, it should be replaced with "="